src/jobs/keys-update/keys-update.service.ts

```diff
@@ -90,119 +86,144 @@ export class KeysUpdateService {
-  }
-
   /**
    * Collects updates from the registry contract and saves the changes to the database
    */
   @OneAtTime()
   private async updateKeys(): Promise<void> {
     await this.jobService.wrapJob({ name: 'Update Staking Router Modules keys' }, async () => {
-      // get blockHash for 'latest' block
-      const blockHash = await this.executionProvider.getBlockHash('latest');
-
-      // get staking router modules
-      const modules = await this.stakingRouterFetchService.getStakingModules({ blockHash: blockHash });
-      this.stakingModules = modules;
-
-      // Here should be a transaction in future that will wrap updateKeys calls of all modules
-      // or other way to call updateKeys method consistently
-
-      await Promise.all(
-        this.stakingModules.map(async (stakingModule) => {
-          if (stakingModule.type === STAKING_MODULE_TYPE.CURATED_ONCHAIN_V1_TYPE) {
-            this.logger.log('Start updating curated keys');
-            await this.curatedModuleService.updateKeys(blockHash);
-          }
-        }),
-      );
-
-      // Update cached data to quick access
-      await this.updateMetricsCache();
+      const meta = await this.update();
+      await this.updateMetrics();
 
-      this.setMetrics();
+      if (meta) {
+        this.lastBlockNumber = meta.number;
+        this.lastTimestampSec = meta.timestamp;
+      }
 
-      // update finished
       // clear timeout
       this.checkKeysUpdateTimeout();
     });
   }
 
   /**
-   * Update cache data above
+   * Update keys of staking modules
+   * @returns Number, hash and timestamp of execution layer block
    */
-  @Trace(METRICS_TIMEOUT)
-  private async updateMetricsCache() {
-    const meta = await this.updateCuratedMetricsCache();
-    // update meta
-    // timestamp and block number is common for all modules
-    this.lastTimestampSec = meta?.timestamp ?? this.lastTimestampSec;
-    this.lastBlockNumber = meta?.blockNumber ?? this.lastBlockNumber;
-  }
+  public async update(): Promise<{ number: number; hash: string; timestamp: number } | undefined> {
+    // reading latest block from blockchain
+    const currElMeta = await this.executionProvider.getBlock('latest');
+    // read from database last execution layer data
+    const prevElMeta = await this.elMetaStorage.get();
+
+    if (prevElMeta && prevElMeta?.blockNumber > currElMeta.number) {
+      this.logger.warn('Previous data is newer than current data', prevElMeta);
+      console.log('curr', currElMeta);
+      return;
+    }
 
-  private async updateCuratedMetricsCache(): Promise<RegistryMeta | null> {
-    const { operators, meta } = await this.curatedModuleService.getOperatorsWithMeta();
-    // should set for each module
-    this.curatedNonce = meta?.keysOpIndex ?? this.curatedNonce;
-    // update curated module operators map
-    this.curatedOperators = operators;
+    // TODO: еcли была реорганизация, может ли currElMeta.number быть меньше и нам надо обновиться ?
```

Comment:
in eng please

Reply:
better to delete this comment , but i need to find answer on question 
Is this condition from `registry library` is correct? 

```
   if (prevElMeta && prevElMeta?.blockNumber > currElMeta.number) {
      this.logger.warn('Previous data is newer than current data', prevElMeta);
      return;
   }
```

Reply:
this condition is correct